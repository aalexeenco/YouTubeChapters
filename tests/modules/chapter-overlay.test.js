import YTPlayerChapterOverlayElement from "./../../js/modules/chapter-overlay.mjs";

test("Chapter overlay element contains empty overlay container when chapter title is not set", () => {
    const element = new YTPlayerChapterOverlayElement();
    expect(element.innerHTML).toMatchSnapshot();
});

test("Chapter overlay element contains empty overlay container when chapter title is set to ''", () => {
    const element = new YTPlayerChapterOverlayElement();
    element.chapterTitle = "";
    expect(element.innerHTML).toMatchSnapshot();
});

test("Chapter overlay element with chapter title 'Hello, World!'", () => {
    const element = new YTPlayerChapterOverlayElement();
    element.chapterTitle = "Hello, World!";
    expect(element.innerHTML).toMatchSnapshot();
});

test.each(["Chapter1", "Chapter2", "ChapterTitle", ""])(
    "Setting chapter title to '%s' removes overlay node to trigger css animation",
    (newChapterTitleText, done) => {
        const element = new YTPlayerChapterOverlayElement();
        const chapterTitleText = "ChapterTitle";
        element.chapterTitle = chapterTitleText;

        const observer = new MutationObserver((mutations, thisObserver) => {
            if (mutations[0].removedNodes[0]?.textContent === chapterTitleText) {
                thisObserver.disconnect();
                done();
            }
        });
        observer.observe(element, { childList: true, subtree: true });

        element.chapterTitle = newChapterTitleText;
    }
);
