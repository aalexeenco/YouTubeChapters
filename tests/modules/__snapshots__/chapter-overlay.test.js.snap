// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapter overlay element contains empty overlay container when chapter title is not set 1`] = `"<div class=\\"ytp-player-content ytp-chapter-title-overlay-container\\"></div>"`;

exports[`Chapter overlay element contains empty overlay container when chapter title is set to '' 1`] = `"<div class=\\"ytp-player-content ytp-chapter-title-overlay-container\\"></div>"`;

exports[`Chapter overlay element with chapter title 'Hello, World!' 1`] = `"<div class=\\"ytp-player-content ytp-chapter-title-overlay-container\\"><span class=\\"ytp-chapter-title-overlay ytp-overlay-reveal\\">Hello, World!</span></div>"`;
